com.unity.ml-agents/Tests/Editor/MLAgentsEditModeTest.cs

```diff
@@ -518,30 +516,44 @@ public void TestMaxStepsReset()
             decisionRequester.DecisionPeriod = 1;
             decisionRequester.Awake();
 
-            var maxStep = 6;
+            const int maxStep = 6;
             agent1.maxStep = maxStep;
             agent1.LazyInitialize();
 
-            int expectedResets= 0;
-            int expectedAgentAction = 0;
-            int expectedAgentActionSinceReset = 0;
+            var expectedAgentStepCount = 0;
+            var expectedResets= 0;
+            var expectedAgentAction = 0;
+            var expectedAgentActionSinceReset = 0;
+            var expectedCollectObsCalls = 0;
+            var expectedCollectObsCallsSinceReset = 0;
 
             for (var i = 0; i < 15; i++)
             {
+                // Agent should observe and act on each Academy step
                 expectedAgentAction += 1;
                 expectedAgentActionSinceReset += 1;
-                if (expectedAgentActionSinceReset == maxStep || (i == 0)){
+                expectedCollectObsCalls += 1;
+                expectedCollectObsCallsSinceReset += 1;
+                expectedAgentStepCount += 1;
+
+                // If the next step will put the agent at maxSteps, we expect it to reset
+                if (agent1.GetStepCount() == maxStep - 1 || (i == 0)){
```

Comment:
IMHO this is a little clearer (since it's a value on the actual Agent, not just a test variable)